File: src/main/scala/s/template_method/like/functional/Display.scala
package s.template_method.like.functional

final class Display(
                     opener: () => Unit,
                     printer: () => Unit,
                     closer: () => Unit
) {
  def display(): Unit = {
    opener()
    for (_ <- 0 until 5) {
      printer()
    }
    closer()
  }
}

File: src/main/kotlin/k/template_method/like/functions/Display.kt
package k.template_method.like.functions

class Display(
    private val opener: () -> Unit,
    private val printer: () -> Unit,
    private val closer: () -> Unit
) {
    fun display() {
        opener()
        repeat(5) {
            printer()
        }
        closer()
    }
}